hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieDefaultTimeline.java

```diff
@@ -629,16 +649,23 @@ private Set<String> getOrCreatePendingClusteringInstantSet() {
   /**
    * Returns the first non savepoint commit on the timeline.
    */
-  private static Option<HoodieInstant> findFirstNonSavepointCommit(List<HoodieInstant> instants) {
+  private static Option<HoodieInstant> findFirstNonSavepointCommit(
+      List<HoodieInstant> instants,
+      boolean byCompletionTime) {
```

Comment:
Can we pass around a func like `() -> HoodieInstant:getCompletionTime` instead of a flag.

Reply:
Fixed.  I also improved the logic to be simpler.